tests/std/json.test.luau

```diff
@@ -55,11 +55,29 @@ test.suite("JSONParsingTestSuite", function(suite)
 		assert.eq(type(obj), "table")
 		assert.eq(obj.name, "Alice")
 		assert.eq(obj.age, 30)
+	end)
+
+	suite:case("json_asobject_test", function(assert)
+		local myjsonstr = [[
+		{
+			"a": true,
+			"b": {
+				"c": true
+			}
+		}
```

Comment:
one last comment - can you extend this test case so that there is an array keyed by "d" inside of the table "b"? Can you show that iterating over the array produces the correct keys.